instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py

```diff
@@ -690,19 +690,23 @@ async def otel_send(message: dict[str, Any]):
                         duration_attrs[
                             SpanAttributes.HTTP_STATUS_CODE
                         ] = status_code
-                        set_status_code(server_span, status_code)
                         set_status_code(send_span, status_code)
 
                         expecting_trailers = message.get("trailers", False)
                     elif message["type"] == "websocket.send":
-                        set_status_code(server_span, 200)
                         set_status_code(send_span, 200)
                     send_span.set_attribute("type", message["type"])
-                    if (
-                        server_span.is_recording()
-                        and server_span.kind == trace.SpanKind.SERVER
-                        and "headers" in message
+
```

Comment:
Why is OTel even setting a `status_code` for WebSockets?

Reply:
I don't know, but I did not want to introduce a breaking change for this fix, but opted to keep the existing behavior.